src/suites/cts/validation/error_scope.spec.ts

```diff
@@ -0,0 +1,128 @@
+export const description = `
+error scope validation tests.
+`;
+
+import { getGPU } from '../../../framework/gpu/implementation.js';
+import { Fixture, TestGroup } from '../../../framework/index.js';
+
+class F extends Fixture {
+  device: GPUDevice = undefined!;
+
+  async init(): Promise<void> {
+    super.init();
+    const gpu = getGPU();
+    const adapter = await gpu.requestAdapter();
+    this.device = await adapter.requestDevice();
+  }
+
+  createErrorBuffer(): void {
+    this.device.createBuffer({
+      size: 1024,
+      usage: 0xffff, // Invalid GPUBufferUsage
+    });
+    // TODO: Remove when chrome does it automatically.
+    this.device.getQueue().submit([]);
+  }
+
+  async expectUncapturedError(fn: Function): Promise<void> {
+    // TODO: Make arbitrary timeout value a test runner variable
+    const TIMEOUT_IN_MS = 1000;
+
+    return this.asyncExpectation(async () => {
+      try {
+        const promise = new Promise((resolve, reject) => {
+          const timeoutId = window.setTimeout(() => {
+            reject(new Error('Uncaptured error timeout occured'));
+          }, TIMEOUT_IN_MS);
+
+          const onUncapturedError = () => {
+            window.clearTimeout(timeoutId);
+            resolve();
+          };
+
+          this.device.addEventListener('uncapturederror', onUncapturedError, { once: true });
+        });
+
+        fn();
+
+        await promise;
+      } catch (error) {
+        this.fail(error.message);
+      }
+    });
```

Comment:
nice! Perhaps `Promise.race` would make this a little cleaner:

```typescript
function rejectTimeout(ms: number, msg: string) -> Promise<void> {
  return new Promise((resolve, reject) => {
    setTimeout(() => { reject(new Error(msg)); }, ms);
  });
}

...

  const promise = new Promise(resolve => {
    this.device.addEventListener('uncapturederror', resolve, { once: true });
  });
  await Promise.race([promise, rejectTimeout(TIMEOUT_IN_MS, 'Uncaptured error timeout occurred')]);
```

Reply:
So much better! Thanks!